core/ecschema-editing/src/Merging/CAClassMerger.ts

```diff
@@ -0,0 +1,28 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { CustomAttributeClass, parseCustomAttributeContainerType } from "@itwin/ecschema-metadata";
+import { MutableCAClass } from "../Editing/Mutable/MutableCAClass";
+import { ClassChanges } from "../Validation/SchemaChanges";
+import { mergeSchemaItemProperties } from "./SchemaItemMerger";
+import mergeClasses from "./ClassMerger";
+
+/**
+ * @internal
+ */
+export default async function mergeCAClasses(target: CustomAttributeClass, source: CustomAttributeClass, changes: ClassChanges) {
+  const mutableCAClass = target as MutableCAClass;
+  await mergeSchemaItemProperties(mutableCAClass, changes.propertyValueChanges, (item, propertyName, propertyValue) => {
+    switch(propertyName) {
+      case "appliesTo": {
+        const containerType = parseCustomAttributeContainerType(propertyValue);
+        if (containerType !== undefined) {
+          return item.setContainerType(containerType);
```

Comment:
container type is a flags enum so setting it to the new value is not really a 'merge', it would be more correct to add the new flag(s) to the existing ones.  This is OK for now but create a new issue to track proper merging of container types then you can resolve this comment

Reply:
Added Issue: https://github.com/iTwin/itwinjs-core/issues/6014